src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,524 @@
+                final AccountConfigData accConfig = getAccountConfig();
+                final int timeToAdd = accConfig.playerData.get(uuid).timePlayed + time;
+                accConfig.playerData.get(uuid).timePlayed = timeToAdd;
+                accConfig.save();
+                return timeToAdd;
+            }
+        }
+        catch (NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+        return -1;
+    }
+
+    public HashMap<UUID, Integer> getPlayers()
+    {
+        final HashMap<UUID, Integer> uuids = new HashMap<>();
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                final ResultSet results = getQuery("SELECT " + UUID_COLUMN + " from " + table_id);
+                while (results != null && results.next())
+                {
+                    final UUID uuid = UUID.fromString(results.getString(UUID_COLUMN));
+                    uuids.put(uuid, getPlayerTime(uuid));
+                }
+            }
+            else
+            {
+                final AccountConfigData accConfig = getAccountConfig();
+                for (final UUID uuid : accConfig.playerData.keySet())
+                {
+                    uuids.put(uuid, getPlayerTime(uuid));
+                }
+            }
+        }
+        catch (NoModuleException | SQLException e)
+        {
+            e.printStackTrace();
+        }
+        return uuids;
+    }
+
+    public boolean doesPlayerExist(final UUID uuid)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                Connection conn = getConn();
+
+                if (conn != null)
+                {
+                    final ResultSet results = getQuery("SELECT " + PLAYER_NAME_COLUMN + " FROM " + table_id + " WHERE " + UUID_COLUMN + " = '" + uuid + "'");
+
+                    if (results != null)
+                    {
+                        return results.next();
+                    }
+                }
+            }
+            else
+            {
+                return getAccountConfig().playerData.containsKey(uuid);
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public void addPlayer(final UUID uuid)
+    {
+        final User user = Sponge.getServiceManager().provideUnchecked(UserStorageService.class).get(uuid).orElse(null);
+
+        if (user == null || doesPlayerExist(uuid))
+        {
+            return;
+        }
+
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                Connection conn = getConn();
+
+                if (conn != null)
+                {
+                    Statement stmt = conn.createStatement();
+
+                    stmt.execute("INSERT INTO " + table_id
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use a variable binding mechanism to construct this query instead of concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2077)